### PR TITLE
Adding privilege checks for binlog commands

### PR DESCRIPTION
**Pull request by fulghum**
Added support for binlog replication privilege checks, tests to assert dynamic privileges are only used at a global scope, and opChecker support for testing dynamic privileges.

### CODE REVIEW

**Review by Hydrocharged — COMMENTED** [review]
Looks good regarding the direction! I dig it 😎

**Comment by Hydrocharged** [comment]
Looks like you just made it an error to try and apply a dynamic privilege at the non-global level, so you can ignore that part in my feedback. What timing lol

File: sql/plan/replication_commands.go [hunk 1]
```diff
@@ -18,10 +18,18 @@ import (
...
+
 	"github.com/dolthub/go-mysql-server/sql"
 	"github.com/dolthub/go-mysql-server/sql/binlogreplication"
 )
 
+// ErrNoReplicationController is returned when replication commands are executed without a configured
+// replication controller to dispatch the command to.
+var ErrNoReplicationController = errors.NewKind("no replication controller available")
+
+const DynamicPrivilge_ReplicationSlaveAdmin = "replication_slave_admin"
```
[the remark this answers is not in the document]
Comment (fulghum) [reply sql/plan/replication_commands.go:33]: Good catch!

**Comment by fulghum** [comment]
Thanks @Hydrocharged! I just finished polishing up the last details, so should be ready for one more quick look from ya.

**Review by Hydrocharged — APPROVED** [review]
LGTM! 🚢

**Pull request merged**